### PR TITLE
[AIOPS-32] Adding in support, tests, and documentation for Docker Compose

### PR DESCRIPTION
**Pull request by kdendtler**
…Docker Compose Libraries

# PR Details

Adding in library support for  Docker Compose. 

## Description

Added library for Docker Compose that supports up and down commands with multiple files, environment variables, and the ability to pause the pipeline for some amount of time to ensure the containers have had time to deploy before continuing.

## How Has This Been Tested

Tested on local Jenkins deployment using all parameters to deploy containers, execute integration tests against those containers, and tear down when done.
Automated tests written and all existing and new tests pass.

## Types of Changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I am submitting this pull request to the appropriate branch
- [x] I have labeled this pull request appropriately
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: libraries/docker_compose/compose.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package libraries.docker_compose
+
+void call(){
+
+}
```
Comment (steven-terrana) [inline libraries/docker_compose/compose.groovy:5]:
```suggestion
```

File: libraries/docker_compose/compose.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+void up() {
+    stage("Deploy") {
```
Comment (steven-terrana) [inline libraries/docker_compose/compose.groovy:8]:
```suggestion
    stage("Deploy: Docker Compose") {
```

File: libraries/docker_compose/compose.groovy [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        sh command
+
+        if (config.sleep) {
+            sleep time: config.sleep.time, unit: config.sleep.unit
+        }
+    }
+}
+
+void down() {
+    stage("Teardown") {
```
Comment (steven-terrana) [inline libraries/docker_compose/compose.groovy:23]:
```suggestion
    stage("Teardown: Docker Compose") {
```

**Review by steven-terrana — APPROVED** [review]

**Comment by kdendtler** [comment]
Declining in favor of the PR to main.

**Pull request closed without merging**